### DISCUSSION TITLE
Related to February end 2024 data submission of waveforms/numerics few questions

### DISCUSSION THREAD

**Discussion by RGanta** [chorus-ai/chorus_waveform#8] [General]
**Related to Feb. 2024 end submission for waveforms/numerics(vitals), questions from NCH site:**

1. For Waveforms: What are the channels/signals (wave labels) need to limit to? Do we need to send all channels/signals (wave labels) that is collected for each patient? 
   For numerics (vitals): What vitals (labels) need to limit to and pulled for Feb. 2024 submission?
2.  For Waveforms and Numerics(Vitals): related to LOS >= 30 days
     For example: (a) If a patient (patient123) is admitted on Jan 1st, 2023 and got discharged on Feb. 15th, 2023. In this case you are asking us to collect whatever waveforms and numerics (vitals) is recorded, collect the data from Day 30 (Jan. 30th, 2023) to  Day 45 (Feb. 15th, 2023) right?
  (b) If same patient (patient123) is admitted again on June 1st, 2023 and got discharged on Aug. 25th, 2023. Are you okay if we generate whatever waveforms and numerics (vitals) is recorded, collect the data from Day 30 (July 1st, 2023)  to Day .. (August 25th, 2023) right? 
3. Regarding number of files for each patientid, are you okay for patinentid(s) generated with files like this for example:
    patient123_01302023_02152023_Waveforms.parquet  (MMDDYYYY)
    patient123_01302023_02152023_Numerics.parquet
    
   patient123_07012023_08252023_Waveforms.parquet
   patient123_07012023_08252023_Numerics.parquet
    
   (OR) for patient123 the above two waveforms file need to be combined into one file? if yes, what should be the file named like ?
  
  patient456_10302023_11152023_Waveforms.parquet
  patient456_10302023_11152023_Numerics.parquet
  
4.  Naming of file for waveforms and numerics (vitals). Example: [PatientId_StartEpoch_EndEpoch_Waveform.parquet/wfdb/h5/csv], [PatientId_StartEpoch_EndEpoch_Numeric.parquet/wfdb/h5/csv] ? Here storing the StartTimestamp in Epoch (StartEpoch) and EndTimestamp in Epoch (EndEpoch) for that PatientId.
5.  a. For sites with waveforms and numeric (vitals) data is being collected from Philips monitor( NCH, (Seattle Childrens)): What fields need to be pulled for waveforms if in parquet: (PatientID, Timestamp (OR) [in Epoch format?], WaveSample (binary hexdec ==example field value: 0xFFADFAHADK000223........), WaveLabel). Are you okay with the following fields in Waveforms file?
    b.  For numerics (vitals), the following fields will be pulled in parquet: (PatientID, Timestamp (OR) [in Epoch format?], NumericLabel, NumericValue). Are you okay with the following fields in numeric file?

**Comment by wa6gz** [comment]
1) All is preferred, understanding that file sizes can be large but it should be possible with Parquet.
2) We are asking for the first 30 days - it would be a) Jan 1 to Jan 30, and b) Jun 1 to Jun 30.
3 and 4) You can combine into one file, or split files, numbering using the standard naming convention with an additional numeric index (patient123_01302023_02152023_3600_0001, ..._0002, etc.)
5) We would need all four of each field to reconstruct the data, so please include all of them. Timestamp in datetime or epoch time is fine.

**Comment by del42** [comment]
@RGanta  @wa6gz  these naming format doesn't reflect the SOP instruction [here](https://chorus-ai.github.io/data_acq_SOP/docs/Data-Uploading/Data_uploading#the-waveforms-folder) 
Didn't we agree on UTC? now Epoch? 
Would you also want sites to separate all numerics and waveforms? 

@clermontg cc'ing you incase
